### PR TITLE
First attempt at adding user authentication using oauth2

### PR DESCRIPTION
**Pull request by Ymihere03**
Here's my attempt to add user authentication and display it in the javascript. It seemed to work ok on the local server using the example@example.com user, but I got a NOT_ALLOWED error when it ran from the hosted appengine.

### CODE REVIEW

**Comment by redmanatee** [comment]
So I think I might not have been clear - rather than using the oAuth libraries directly, I would think you would want to use the [GAE User APIs](https://cloud.google.com/appengine/docs/go/users/).  They should handle a lot of the oAuth nastiness, and hopefully make it integrate in easier.

**Comment by Ymihere03** [comment]
Hey, I fixed everything up including the way I was integrating the UI. Login and logout should both work now.

File: js/stores/user.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		if(this.user.Url.indexOf('http') == -1) {
+			window.location.href = 'localhost:8080'+this.user.Url;
+			return;
+		}
+		
+		window.location.href = this.user.Url;
+	}
+});
+	
+
```
Comment (redmanatee) [inline js/stores/user.js]: Some weird whitespace at the end of this file.  Should have one blank line after the end of the file.

File: app.yaml [hunk 2]
```diff
@@ -19,6 +19,7 @@ handlers:
   login: admin
 - url: /.*
   script: _go_app
+  login: required
```
Comment (redmanatee) [inline app.yaml]: This is probably OK for the current pass, but I think I will want people to be able to browse the site without having to be logged in.

**Pull request merged**